GPU/GLES/Framebuffer.cpp

```diff
@@ -667,8 +667,44 @@ void FramebufferManager::ResizeFramebufFBO(VirtualFramebuffer *vfb, u16 w, u16 h
 		vfb->bufferHeight = std::max(vfb->bufferHeight, h);
 	}
 
-	vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
-	vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+	if (g_Config.iInternalResolution==1) {
+		vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
+		vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+	} else {
+		switch (g_Config.iBloomHack) {
+		case 0:
+			vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
+			vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+			break;
+		case 1:
+			if (vfb->bufferWidth <= 128 || vfb->bufferHeight <= 64) {
+				vfb->renderWidth = vfb->bufferWidth;
+				vfb->renderHeight = vfb->bufferHeight;
+			} else {
+				vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
+				vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+			}
+			break;
+		case 2:
+			if (vfb->bufferWidth <= 256 || vfb->bufferHeight <= 128) {
+				vfb->renderWidth = vfb->bufferWidth;
+				vfb->renderHeight = vfb->bufferHeight;
+			} else {
+				vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
+				vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+			}
+			break;
+		case 3:
+			if (vfb->bufferWidth < 480 || vfb->bufferHeight < 272) {
+				vfb->renderWidth = vfb->bufferWidth;
+				vfb->renderHeight = vfb->bufferHeight;
+			} else {
+				vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
+				vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
+			}
+			break;
+		}
+	}
```

Comment:
Maybe this ought to be in some method on FramebufferCommon...

Anyway, I would structure it more like:

```c++
bool force1x = false;
switch (g_Config.iBloomHack) {
case 1:
   force1x = vfb->bufferWidth <= 128 || vfb->bufferHeight <= 64;
   break;
case 2:
   force1x = vfb->bufferWidth <= 256 || vfb->bufferHeight <= 128;
   break;
case 3:
   force1x = vfb->bufferWidth <= 480 || vfb->bufferHeight <= 272;
   break;
}

if (force1x && g_Config.iInternalResolution != 1) {
   vfb->renderWidth = vfb->bufferWidth;
   vfb->renderHeight = vfb->bufferHeight;
} else {
   vfb->renderWidth = vfb->bufferWidth * renderWidthFactor;
   vfb->renderHeight = vfb->bufferHeight * renderHeightFactor;
}
```

Either way, `g_Config.iInternalResolution==1` should have spaces.

Also, I wonder if `&&` for w/h would be better in some cases but doubt it matters...

-[Unknown]